Comparison Engine/Comparison Engine/Form1.cs

```diff
@@ -184,72 +172,75 @@ private void configureChildForm(Form childForm)
             childForm.BringToFront();
             childForm.Show();
         }
-        public void closeActiveForm()
+        public void CloseActiveForm()
         {
             activeForm.Close();
         }
 
         //LIST BUTTON CREATOR #listbuttons
 
         //BAR BUTTON FUNCTIONS
-        private void createBarButton(Bar bar)
+        private void CreateBarButton(Bar bar)
         {
             Button button = new Button();
-            configureBarButton(button, bar);
-            bindBar(button, bar);
+            ConfigureBarButton(button, bar);
+            BindBar(button, bar);
         }
 
-        private void bindBar(Button button, Bar bar)
+        private void BindBar(Button button, Bar bar)
         {
-            button.Click += (sender, EventArgs) => { barButtonClick(bar); };
+            button.Click += (sender, EventArgs) => { BarButtonClick(bar); };
         }
 
-        private void barButtonClick(Bar bar)
+        private void BarButtonClick(Bar bar)
         {
-            openChildFormBar(bar);
+            OpenChildFormBar(bar);
+            mapController.RemoveOverlays(map);
+            mapController.ShowRoute(map, bar.barLocation);
         }
-        private void configureBarButton(Button button, Bar bar)
+        private void ConfigureBarButton(Button button, Bar bar)
         {
             panelSubList.Controls.Add(button);
             button.Text = (bar.barName);
             button.Dock = DockStyle.Top;
         }
         //DRINK BUTTON FUNCTIONS
-        private void createDrinkButton(Drink drink)
+        private void CreateDrinkButton(Drink drink)
         {
             Button button = new Button();
-            configureDrinkButton(button, drink);
-            bindDrink(button, drink);
+            ConfigureDrinkButton(button, drink);
+            BindDrink(button, drink);
         }
 
-        private void bindDrink(Button button, Drink drink)
+        private void BindDrink(Button button, Drink drink)
         {
-            button.Click += (sender, EventArgs) => { drinkButtonClick(drink); };
+            button.Click += (sender, EventArgs) => { DrinkButtonClick(drink); };
         }
 
-        private void drinkButtonClick(Drink drink)
+        private void DrinkButtonClick(Drink drink)
         {
-           openChildFormDrink(drink);
-           // mapController.ShowBarsWithDrink(map, drink, barManager.barDictionary);
```

Comment:
I left this commented during testing to avoid exceptions. This should probably be uncommented now.

Reply:
On my end the line is uncommented... I'm gonna pretend that it's ok